lib/commands/audit.js

```diff
@@ -1,8 +1,389 @@
 const Arborist = require('@npmcli/arborist')
 const auditReport = require('npm-audit-report')
-const reifyFinish = require('../utils/reify-finish.js')
-const auditError = require('../utils/audit-error.js')
+const chalk = require('chalk')
+const crypto = require('crypto')
+const fetch = require('npm-registry-fetch')
+const localeCompare = require('@isaacs/string-locale-compare')('en')
+const npa = require('npm-package-arg')
+const pacote = require('pacote')
+
 const ArboristWorkspaceCmd = require('../arborist-cmd.js')
+const auditError = require('../utils/audit-error.js')
+const {
+  registry: { default: defaultRegistry },
+} = require('../utils/config/definitions.js')
+const log = require('../utils/log-shim.js')
+const pulseTillDone = require('../utils/pulse-till-done.js')
+const reifyFinish = require('../utils/reify-finish.js')
+
+const verifySignature = async ({ message, signature, publicKey }) => {
+  const verifier = crypto.createVerify('SHA256')
+  verifier.write(message)
+  verifier.end()
+  return verifier.verify(publicKey, signature, 'base64')
+}
+
+class VerifySignatures {
+  constructor (tree, filterSet, npm, opts) {
+    this.tree = tree
+    this.filterSet = filterSet
+    this.npm = npm
+    this.opts = opts
+    this.edges = new Set()
+    this.keys = new Map()
+    this.invalid = new Set()
+    this.missing = new Set()
+    this.audited = 0
+    this.verified = 0
+    this.output = []
+    this.exitCode = 0
+  }
+
+  async run () {
+    const start = process.hrtime.bigint()
+
+    // Find all deps in tree
+    const nodes = this.tree.inventory.values()
+    this.getEdges(nodes, 'edgesOut')
+    const edges = Array.from(this.edges)
+    if (edges.length === 0) {
```

Comment:
```suggestion
    if (this.edges.size === 0) {
```
that way you can pass the Set around directly?